{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NAXE features:\n",
      "                  chromosome                 source feature_type      start  \\\n",
      "241841   chr1_MU273335v1_fix  ncbiRefSeq.2022-10-28   transcript     150333   \n",
      "241842   chr1_MU273335v1_fix  ncbiRefSeq.2022-10-28         exon     150333   \n",
      "241843   chr1_MU273335v1_fix  ncbiRefSeq.2022-10-28         5UTR     150333   \n",
      "241844   chr1_MU273335v1_fix  ncbiRefSeq.2022-10-28          CDS     150362   \n",
      "241845   chr1_MU273335v1_fix  ncbiRefSeq.2022-10-28         exon     150658   \n",
      "241846   chr1_MU273335v1_fix  ncbiRefSeq.2022-10-28          CDS     150658   \n",
      "241847   chr1_MU273335v1_fix  ncbiRefSeq.2022-10-28         exon     150922   \n",
      "241848   chr1_MU273335v1_fix  ncbiRefSeq.2022-10-28          CDS     150922   \n",
      "241849   chr1_MU273335v1_fix  ncbiRefSeq.2022-10-28         exon     151114   \n",
      "241850   chr1_MU273335v1_fix  ncbiRefSeq.2022-10-28          CDS     151114   \n",
      "241851   chr1_MU273335v1_fix  ncbiRefSeq.2022-10-28         exon     151965   \n",
      "241852   chr1_MU273335v1_fix  ncbiRefSeq.2022-10-28          CDS     151965   \n",
      "241853   chr1_MU273335v1_fix  ncbiRefSeq.2022-10-28         exon     152439   \n",
      "241854   chr1_MU273335v1_fix  ncbiRefSeq.2022-10-28          CDS     152439   \n",
      "241855   chr1_MU273335v1_fix  ncbiRefSeq.2022-10-28         3UTR     152642   \n",
      "241856   chr1_MU273335v1_fix  ncbiRefSeq.2022-10-28  start_codon     150362   \n",
      "241857   chr1_MU273335v1_fix  ncbiRefSeq.2022-10-28   stop_codon     152639   \n",
      "4571171                 chr1  ncbiRefSeq.2022-10-28   transcript  156591776   \n",
      "4571172                 chr1  ncbiRefSeq.2022-10-28         exon  156591776   \n",
      "4571173                 chr1  ncbiRefSeq.2022-10-28         5UTR  156591776   \n",
      "4571174                 chr1  ncbiRefSeq.2022-10-28          CDS  156591805   \n",
      "4571175                 chr1  ncbiRefSeq.2022-10-28         exon  156592101   \n",
      "4571176                 chr1  ncbiRefSeq.2022-10-28          CDS  156592101   \n",
      "4571177                 chr1  ncbiRefSeq.2022-10-28         exon  156592365   \n",
      "4571178                 chr1  ncbiRefSeq.2022-10-28          CDS  156592365   \n",
      "4571179                 chr1  ncbiRefSeq.2022-10-28         exon  156592557   \n",
      "4571180                 chr1  ncbiRefSeq.2022-10-28          CDS  156592557   \n",
      "4571181                 chr1  ncbiRefSeq.2022-10-28         exon  156593408   \n",
      "4571182                 chr1  ncbiRefSeq.2022-10-28          CDS  156593408   \n",
      "4571183                 chr1  ncbiRefSeq.2022-10-28         exon  156593882   \n",
      "4571184                 chr1  ncbiRefSeq.2022-10-28          CDS  156593882   \n",
      "4571185                 chr1  ncbiRefSeq.2022-10-28         3UTR  156594085   \n",
      "4571186                 chr1  ncbiRefSeq.2022-10-28  start_codon  156591805   \n",
      "4571187                 chr1  ncbiRefSeq.2022-10-28   stop_codon  156594082   \n",
      "\n",
      "               end score strand frame gene_id  transcript_id gene_name  \\\n",
      "241841      152856     .      +     .    NAXE    NM_144772.3      NAXE   \n",
      "241842      150543     .      +     .    NAXE    NM_144772.3      NAXE   \n",
      "241843      150361     .      +     .    NAXE    NM_144772.3      NAXE   \n",
      "241844      150543     .      +     0    NAXE    NM_144772.3      NAXE   \n",
      "241845      150766     .      +     .    NAXE    NM_144772.3      NAXE   \n",
      "241846      150766     .      +     1    NAXE    NM_144772.3      NAXE   \n",
      "241847      151032     .      +     .    NAXE    NM_144772.3      NAXE   \n",
      "241848      151032     .      +     0    NAXE    NM_144772.3      NAXE   \n",
      "241849      151227     .      +     .    NAXE    NM_144772.3      NAXE   \n",
      "241850      151227     .      +     0    NAXE    NM_144772.3      NAXE   \n",
      "241851      152112     .      +     .    NAXE    NM_144772.3      NAXE   \n",
      "241852      152112     .      +     0    NAXE    NM_144772.3      NAXE   \n",
      "241853      152856     .      +     .    NAXE    NM_144772.3      NAXE   \n",
      "241854      152638     .      +     2    NAXE    NM_144772.3      NAXE   \n",
      "241855      152856     .      +     .    NAXE    NM_144772.3      NAXE   \n",
      "241856      150364     .      +     0    NAXE    NM_144772.3      NAXE   \n",
      "241857      152641     .      +     0    NAXE    NM_144772.3      NAXE   \n",
      "4571171  156594299     .      +     .    NAXE  NM_144772.3_2      NAXE   \n",
      "4571172  156591986     .      +     .    NAXE  NM_144772.3_2      NAXE   \n",
      "4571173  156591804     .      +     .    NAXE  NM_144772.3_2      NAXE   \n",
      "4571174  156591986     .      +     0    NAXE  NM_144772.3_2      NAXE   \n",
      "4571175  156592209     .      +     .    NAXE  NM_144772.3_2      NAXE   \n",
      "4571176  156592209     .      +     1    NAXE  NM_144772.3_2      NAXE   \n",
      "4571177  156592475     .      +     .    NAXE  NM_144772.3_2      NAXE   \n",
      "4571178  156592475     .      +     0    NAXE  NM_144772.3_2      NAXE   \n",
      "4571179  156592670     .      +     .    NAXE  NM_144772.3_2      NAXE   \n",
      "4571180  156592670     .      +     0    NAXE  NM_144772.3_2      NAXE   \n",
      "4571181  156593555     .      +     .    NAXE  NM_144772.3_2      NAXE   \n",
      "4571182  156593555     .      +     0    NAXE  NM_144772.3_2      NAXE   \n",
      "4571183  156594299     .      +     .    NAXE  NM_144772.3_2      NAXE   \n",
      "4571184  156594081     .      +     2    NAXE  NM_144772.3_2      NAXE   \n",
      "4571185  156594299     .      +     .    NAXE  NM_144772.3_2      NAXE   \n",
      "4571186  156591807     .      +     0    NAXE  NM_144772.3_2      NAXE   \n",
      "4571187  156594084     .      +     0    NAXE  NM_144772.3_2      NAXE   \n",
      "\n",
      "        exon_number          exon_id  \n",
      "241841          NaN              NaN  \n",
      "241842            1    NM_144772.3.1  \n",
      "241843            1    NM_144772.3.1  \n",
      "241844            1    NM_144772.3.1  \n",
      "241845            2    NM_144772.3.2  \n",
      "241846            2    NM_144772.3.2  \n",
      "241847            3    NM_144772.3.3  \n",
      "241848            3    NM_144772.3.3  \n",
      "241849            4    NM_144772.3.4  \n",
      "241850            4    NM_144772.3.4  \n",
      "241851            5    NM_144772.3.5  \n",
      "241852            5    NM_144772.3.5  \n",
      "241853            6    NM_144772.3.6  \n",
      "241854            6    NM_144772.3.6  \n",
      "241855            6    NM_144772.3.6  \n",
      "241856            1    NM_144772.3.1  \n",
      "241857            6    NM_144772.3.6  \n",
      "4571171         NaN              NaN  \n",
      "4571172           1  NM_144772.3_2.1  \n",
      "4571173           1  NM_144772.3_2.1  \n",
      "4571174           1  NM_144772.3_2.1  \n",
      "4571175           2  NM_144772.3_2.2  \n",
      "4571176           2  NM_144772.3_2.2  \n",
      "4571177           3  NM_144772.3_2.3  \n",
      "4571178           3  NM_144772.3_2.3  \n",
      "4571179           4  NM_144772.3_2.4  \n",
      "4571180           4  NM_144772.3_2.4  \n",
      "4571181           5  NM_144772.3_2.5  \n",
      "4571182           5  NM_144772.3_2.5  \n",
      "4571183           6  NM_144772.3_2.6  \n",
      "4571184           6  NM_144772.3_2.6  \n",
      "4571185           6  NM_144772.3_2.6  \n",
      "4571186           1  NM_144772.3_2.1  \n",
      "4571187           6  NM_144772.3_2.6  \n",
      "\n",
      "Gene statistics:\n",
      "{'transcripts': 2, 'exons': 12, 'cds': 12, 'utrs': 4, 'chromosomes': ['chr1_MU273335v1_fix', 'chr1'], 'total_span': np.int64(156443967)}\n",
      "Available transcripts:\n",
      "         transcript_id           chromosome      start        end strand\n",
      "241841     NM_144772.3  chr1_MU273335v1_fix     150333     152856      +\n",
      "4571171  NM_144772.3_2                 chr1  156591776  156594299      +\n",
      "\n",
      "Visualizing transcript: NM_144772.3_2\n"
     ]
    }
   ],
   "source": [
    "from swissisoform.genome import GenomeHandler\n",
    "from swissisoform.visualize import GenomeVisualizer\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "# Initialize the genome handler\n",
    "genome = GenomeHandler(\n",
    "   '../data/genome_data/hg38.fa',\n",
    "   '../data/genome_data/hg38.ncbiRefSeq.gtf'\n",
    ")\n",
    "\n",
    "# Initialize the visualizer\n",
    "visualizer = GenomeVisualizer(genome)\n",
    "\n",
    "# Get NAXE features\n",
    "naxe_features = genome.find_gene_features('NAXE')\n",
    "print(\"NAXE features:\")\n",
    "print(naxe_features)\n",
    "\n",
    "# Get gene statistics\n",
    "stats = genome.get_gene_stats('NAXE')\n",
    "print(\"\\nGene statistics:\")\n",
    "print(stats)\n",
    "\n",
    "# Get available transcript IDs for NAXE\n",
    "transcript_info = genome.get_transcript_ids('NAXE')\n",
    "print(\"Available transcripts:\")\n",
    "print(transcript_info)\n",
    "\n",
    "# Select a transcript ID (for example, the last one)\n",
    "transcript_id = transcript_info.iloc[-1]['transcript_id']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "import pandas as pd\n",
    "from matplotlib.patches import Rectangle\n",
    "from matplotlib.lines import Line2D\n",
    "\n",
    "class GenomeVisualizer:\n",
    "    def __init__(self, genome_handler):\n",
    "        self.genome = genome_handler\n",
    "        self.feature_colors = {\n",
    "            'exon': '#4CAF50',      # green\n",
    "            'CDS': '#2196F3',       # blue\n",
    "            'UTR': '#FFA500',       # orange\n",
    "            'start_codon': '#FF0000', # red\n",
    "            'stop_codon': '#800080',  # purple\n",
    "            'truncation': '#FF1493',  # deep pink\n",
    "            'alternative_start': '#FFD700'  # yellow\n",
    "        }\n",
    "        self.track_height = 0.15\n",
    "        self.codon_height = 0.2\n",
    "\n",
    "    def visualize_transcript(self, gene_name, transcript_id, alt_features=None, output_file=None):\n",
    "        \"\"\"\n",
    "        Visualize transcript with optional alternative start sites.\n",
    "        \n",
    "        Args:\n",
    "            gene_name (str): Name of the gene\n",
    "            transcript_id (str): Transcript ID to visualize\n",
    "            alt_features (pd.DataFrame, optional): Alternative start features\n",
    "            output_file (str, optional): Path to save the visualization\n",
    "        \"\"\"\n",
    "        transcript_data = self.genome.get_transcript_features_with_sequence(transcript_id)\n",
    "        features = transcript_data['features']\n",
    "        \n",
    "        if features.empty:\n",
    "            raise ValueError(f\"No features found for transcript {transcript_id}\")\n",
    "\n",
    "        # Get transcript bounds\n",
    "        transcript_info = features[features['feature_type'] == 'transcript'].iloc[0]\n",
    "        transcript_start = transcript_info['start']\n",
    "        transcript_end = transcript_info['end']\n",
    "        span = transcript_end - transcript_start\n",
    "        \n",
    "        # Create figure\n",
    "        fig, ax = plt.subplots(figsize=(15, 3))\n",
    "        \n",
    "        # Draw base transcript line (thicker)\n",
    "        ax.hlines(y=0.4, xmin=transcript_start, xmax=transcript_end,\n",
    "                 color='black', linewidth=1.5)\n",
    "\n",
    "        # Plot regular transcript features\n",
    "        for _, feature in features.iterrows():\n",
    "            width = feature['end'] - feature['start']\n",
    "            \n",
    "            if feature['feature_type'] == 'exon':\n",
    "                rect = Rectangle((feature['start'], 0.325),\n",
    "                               width,\n",
    "                               self.track_height,\n",
    "                               facecolor=self.feature_colors['exon'],\n",
    "                               alpha=0.3)\n",
    "                ax.add_patch(rect)\n",
    "            \n",
    "            elif feature['feature_type'] == 'CDS':\n",
    "                rect = Rectangle((feature['start'], 0.325),\n",
    "                               width,\n",
    "                               self.track_height,\n",
    "                               facecolor=self.feature_colors['CDS'])\n",
    "                ax.add_patch(rect)\n",
    "            \n",
    "            elif feature['feature_type'] in ['5UTR', '3UTR']:\n",
    "                rect = Rectangle((feature['start'], 0.6),\n",
    "                               width,\n",
    "                               self.track_height,\n",
    "                               facecolor=self.feature_colors['UTR'])\n",
    "                ax.add_patch(rect)\n",
    "            \n",
    "            elif feature['feature_type'] in ['start_codon', 'stop_codon']:\n",
    "                rect = Rectangle((feature['start'], 0.3),\n",
    "                               width,\n",
    "                               self.codon_height,\n",
    "                               facecolor=self.feature_colors[feature['feature_type']])\n",
    "                ax.add_patch(rect)\n",
    "\n",
    "        # Plot alternative start sites if provided\n",
    "        if alt_features is not None and not alt_features.empty:\n",
    "            for _, alt_feature in alt_features.iterrows():\n",
    "                alt_start = alt_feature['start']\n",
    "                \n",
    "                # Draw truncation bracket\n",
    "                alt_end = alt_feature['end']\n",
    "                bracket_height = 0.05  # Height of the vertical parts of bracket\n",
    "                bracket_y = 0.2  # Base y-position of the bracket\n",
    "                \n",
    "                # Draw horizontal line of bracket\n",
    "                ax.hlines(y=bracket_y, \n",
    "                         xmin=alt_start-0.5,\n",
    "                         xmax=alt_end+0.5,\n",
    "                         color=self.feature_colors['truncation'],\n",
    "                         linewidth=1.5,\n",
    "                         label='Truncation')\n",
    "                \n",
    "                # Draw vertical parts of bracket\n",
    "                ax.vlines(x=[alt_start, alt_end-0.5],\n",
    "                         ymin=bracket_y,\n",
    "                         ymax=bracket_y + bracket_height,\n",
    "                         color=self.feature_colors['truncation'],\n",
    "                         linewidth=0.5)\n",
    "                \n",
    "                # Draw alternative start marker (vertical yellow line)\n",
    "                # Calculate vertical position for alternative start marker\n",
    "                black_bar_y = 0.4  # y-position of the black bar\n",
    "                alt_start_height = self.codon_height  # match codon height\n",
    "                alt_start_ymin = black_bar_y - (alt_start_height / 2)  # center around black bar\n",
    "                alt_start_ymax = black_bar_y + (alt_start_height / 2)\n",
    "                \n",
    "                ax.vlines(x=alt_end,\n",
    "                         ymin=alt_start_ymin,\n",
    "                         ymax=alt_start_ymax,\n",
    "                         color=self.feature_colors['alternative_start'],\n",
    "                         linewidth=2,\n",
    "                         label='Alternative start')\n",
    "                \n",
    "                # Add start codon label if available (now above the line)\n",
    "                if 'start_codon' in alt_feature:\n",
    "                    plt.text(alt_end - span*0.01, alt_start_ymax + 0.05, \n",
    "                            alt_feature['start_codon'],\n",
    "                            fontsize=8,\n",
    "                            rotation=45)\n",
    "\n",
    "        # Center title\n",
    "        plt.title(f\"{gene_name} - {transcript_id}\", pad=20, y=1.05)\n",
    "\n",
    "        # Customize plot\n",
    "        ax.set_ylim(0, 1)\n",
    "        ax.set_xlim(transcript_start - 10, transcript_end + 10)\n",
    "        \n",
    "        # Calculate dynamic tick interval based on span\n",
    "        tick_span = transcript_end - transcript_start\n",
    "        if tick_span > 100000:\n",
    "            tick_interval = 10000\n",
    "        elif tick_span > 50000:\n",
    "            tick_interval = 5000\n",
    "        elif tick_span > 10000:\n",
    "            tick_interval = 1000\n",
    "        else:\n",
    "            tick_interval = 500\n",
    "\n",
    "        # Calculate tick positions\n",
    "        base_position = transcript_start - (transcript_start % tick_interval)\n",
    "        tick_positions = range(base_position, transcript_end + tick_interval, tick_interval)\n",
    "        \n",
    "        plt.xticks(tick_positions, [f\"{pos:,}\" for pos in tick_positions])\n",
    "        plt.xticks(rotation=45)\n",
    "        \n",
    "        # Remove y-axis\n",
    "        ax.set_yticks([])\n",
    "\n",
    "        # Add legend without frame\n",
    "        legend_elements = [\n",
    "            plt.Rectangle((0,0), 1, 1, facecolor=self.feature_colors['exon'], \n",
    "                         alpha=0.3, label='Exon'),\n",
    "            plt.Rectangle((0,0), 1, 1, facecolor=self.feature_colors['CDS'], \n",
    "                         label='CDS'),\n",
    "            plt.Rectangle((0,0), 1, 1, facecolor=self.feature_colors['UTR'], \n",
    "                         label='UTR'),\n",
    "            plt.Rectangle((0,0), 1, 1, facecolor=self.feature_colors['start_codon'], \n",
    "                         label='Start codon'),\n",
    "            plt.Rectangle((0,0), 1, 1, facecolor=self.feature_colors['stop_codon'], \n",
    "                         label='Stop codon')\n",
    "        ]\n",
    "        \n",
    "        # Add alternative start elements to legend if relevant\n",
    "        if alt_features is not None and not alt_features.empty:\n",
    "            legend_elements.extend([\n",
    "                Line2D([0], [0], color=self.feature_colors['truncation'], \n",
    "                      label='Truncation', linewidth=1.5),\n",
    "                Line2D([0], [0], color=self.feature_colors['alternative_start'], \n",
    "                      label='Alternative start', linewidth=2)\n",
    "            ])\n",
    "\n",
    "        ax.legend(handles=legend_elements, bbox_to_anchor=(1.05, 1), \n",
    "                 loc='upper left', borderaxespad=0., frameon=False)\n",
    "\n",
    "        plt.tight_layout()\n",
    "        \n",
    "        # Remove spines        \n",
    "        ax.spines['top'].set_visible(False)\n",
    "        ax.spines['right'].set_visible(False)\n",
    "        ax.spines['left'].set_visible(False)\n",
    "\n",
    "        if output_file:\n",
    "            plt.savefig(output_file, bbox_inches='tight', dpi=300,\n",
    "                       facecolor='white', edgecolor='none')\n",
    "            plt.close()\n",
    "        else:\n",
    "            plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from typing import Optional, Dict, List\n",
    "\n",
    "class AlternativeIsoform:\n",
    "    \"\"\"\n",
    "    A class to handle alternative isoform data from BED format files.\n",
    "    The name field in the BED file is expected to contain gene information in the format:\n",
    "    ENSG00000260916.6_CCPG1_AUG_TruncationToAnno\n",
    "    \"\"\"\n",
    "    def __init__(self):\n",
    "        \"\"\"Initialize an empty isoform handler\"\"\"\n",
    "        self.isoforms = pd.DataFrame()\n",
    "        \n",
    "    def load_bed(self, file_path: str) -> None:\n",
    "        \"\"\"\n",
    "        Load data from BED format file.\n",
    "        \n",
    "        Args:\n",
    "            file_path (str): Path to the BED file\n",
    "        \"\"\"\n",
    "        # Read BED format with standard columns\n",
    "        self.isoforms = pd.read_csv(file_path, sep='\\t',\n",
    "                                  names=['chrom', 'start', 'end', 'name', 'score', 'strand'])\n",
    "        \n",
    "        # Parse the name field which contains gene information\n",
    "        def parse_name(name: str) -> Dict[str, str]:\n",
    "            parts = name.split('_')\n",
    "            return {\n",
    "                'gene_id': parts[0],\n",
    "                'gene_name': parts[1],\n",
    "                'start_codon': parts[2],\n",
    "                'isoform_type': parts[3]\n",
    "            }\n",
    "        \n",
    "        # Parse name field and add as new columns\n",
    "        info_df = pd.DataFrame(self.isoforms['name'].apply(parse_name).tolist())\n",
    "        self.isoforms = pd.concat([self.isoforms, info_df], axis=1)\n",
    "        \n",
    "        # Add columns needed for visualization\n",
    "        self.isoforms['feature_type'] = 'alternative_start'\n",
    "        self.isoforms['source'] = 'truncation'\n",
    "\n",
    "    def get_visualization_features(self, gene_name: str) -> pd.DataFrame:\n",
    "        \"\"\"\n",
    "        Get features formatted for the GenomeVisualizer.\n",
    "        \n",
    "        Args:\n",
    "            gene_name (str): Name of the gene to get features for\n",
    "            \n",
    "        Returns:\n",
    "            pd.DataFrame: Features formatted for visualization\n",
    "        \"\"\"\n",
    "        if self.isoforms.empty:\n",
    "            raise ValueError(\"No data loaded. Please load data first.\")\n",
    "            \n",
    "        features = self.isoforms[self.isoforms['gene_name'] == gene_name].copy()\n",
    "        \n",
    "        if features.empty:\n",
    "            return pd.DataFrame()\n",
    "            \n",
    "        # Format features for visualization to match GenomeVisualizer expected format\n",
    "        viz_features = pd.DataFrame({\n",
    "            'chromosome': features['chrom'],\n",
    "            'source': 'truncation',\n",
    "            'feature_type': 'alternative_start',\n",
    "            'start': features['start'],\n",
    "            'end': features['end'],\n",
    "            'score': features['score'],\n",
    "            'strand': features['strand'],\n",
    "            'frame': '.',\n",
    "            'gene_id': features['gene_id'],\n",
    "            'transcript_id': features['gene_id'] + '_alt',\n",
    "            'gene_name': features['gene_name'],\n",
    "            'start_codon': features['start_codon']\n",
    "        })\n",
    "        \n",
    "        return viz_features\n",
    "\n",
    "    def get_gene_list(self) -> List[str]:\n",
    "        \"\"\"\n",
    "        Get list of all genes in the dataset.\n",
    "        \n",
    "        Returns:\n",
    "            List[str]: List of gene names\n",
    "        \"\"\"\n",
    "        if self.isoforms.empty:\n",
    "            raise ValueError(\"No data loaded. Please load data first.\")\n",
    "            \n",
    "        return sorted(self.isoforms['gene_name'].unique().tolist())\n",
    "\n",
    "    def get_stats(self) -> Dict:\n",
    "        \"\"\"\n",
    "        Get basic statistics about the loaded isoforms.\n",
    "        \n",
    "        Returns:\n",
    "            Dict: Dictionary containing various statistics\n",
    "        \"\"\"\n",
    "        if self.isoforms.empty:\n",
    "            raise ValueError(\"No data loaded. Please load data first.\")\n",
    "            \n",
    "        stats = {\n",
    "            'total_sites': len(self.isoforms),\n",
    "            'unique_genes': len(self.get_gene_list()),\n",
    "            'chromosomes': sorted(self.isoforms['chrom'].unique().tolist()),\n",
    "            'start_codons': sorted(self.isoforms['start_codon'].unique().tolist())\n",
    "        }\n",
    "        \n",
    "        return stats\n",
    "\n",
    "    def get_alternative_starts(self, gene_name: Optional[str] = None) -> pd.DataFrame:\n",
    "        \"\"\"\n",
    "        Get alternative start sites, optionally filtered by gene name.\n",
    "        \n",
    "        Args:\n",
    "            gene_name (str, optional): Gene name to filter by\n",
    "            \n",
    "        Returns:\n",
    "            pd.DataFrame: Alternative start sites with positions and metadata\n",
    "        \"\"\"\n",
    "        if self.isoforms.empty:\n",
    "            raise ValueError(\"No data loaded. Please load data first.\")\n",
    "            \n",
    "        if gene_name is not None:\n",
    "            return self.isoforms[self.isoforms['gene_name'] == gene_name]\n",
    "        return self.isoforms.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [],
   "source": [
    "visualizer = GenomeVisualizer(genome)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Initialize handlers\n",
    "alt_isoforms = AlternativeIsoform()\n",
    "\n",
    "# Load BED file\n",
    "alt_isoforms.load_bed('../data/ribosome_profiling/RiboTISHV6_Ly2024_AnnoToTruncation_exonintersect.bed')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [],
   "source": [
    "alt_features = alt_isoforms.get_visualization_features('NAXE')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [],
   "source": [
    "visualizer.visualize_transcript('NAXE', transcript_id, alt_features=None, output_file=f'naxe_transcript_{transcript_id}.png')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [],
   "source": [
    "visualizer.visualize_transcript('NAXE', transcript_id, alt_features=alt_features, output_file=f'naxe_transcript_alt_{transcript_id}.png')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "swissisoform",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.16"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
