{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NAXE features:\n",
      "                  chromosome                 source feature_type      start  \\\n",
      "241841   chr1_MU273335v1_fix  ncbiRefSeq.2022-10-28   transcript     150333   \n",
      "241842   chr1_MU273335v1_fix  ncbiRefSeq.2022-10-28         exon     150333   \n",
      "241843   chr1_MU273335v1_fix  ncbiRefSeq.2022-10-28         5UTR     150333   \n",
      "241844   chr1_MU273335v1_fix  ncbiRefSeq.2022-10-28          CDS     150362   \n",
      "241845   chr1_MU273335v1_fix  ncbiRefSeq.2022-10-28         exon     150658   \n",
      "241846   chr1_MU273335v1_fix  ncbiRefSeq.2022-10-28          CDS     150658   \n",
      "241847   chr1_MU273335v1_fix  ncbiRefSeq.2022-10-28         exon     150922   \n",
      "241848   chr1_MU273335v1_fix  ncbiRefSeq.2022-10-28          CDS     150922   \n",
      "241849   chr1_MU273335v1_fix  ncbiRefSeq.2022-10-28         exon     151114   \n",
      "241850   chr1_MU273335v1_fix  ncbiRefSeq.2022-10-28          CDS     151114   \n",
      "241851   chr1_MU273335v1_fix  ncbiRefSeq.2022-10-28         exon     151965   \n",
      "241852   chr1_MU273335v1_fix  ncbiRefSeq.2022-10-28          CDS     151965   \n",
      "241853   chr1_MU273335v1_fix  ncbiRefSeq.2022-10-28         exon     152439   \n",
      "241854   chr1_MU273335v1_fix  ncbiRefSeq.2022-10-28          CDS     152439   \n",
      "241855   chr1_MU273335v1_fix  ncbiRefSeq.2022-10-28         3UTR     152642   \n",
      "241856   chr1_MU273335v1_fix  ncbiRefSeq.2022-10-28  start_codon     150362   \n",
      "241857   chr1_MU273335v1_fix  ncbiRefSeq.2022-10-28   stop_codon     152639   \n",
      "4571171                 chr1  ncbiRefSeq.2022-10-28   transcript  156591776   \n",
      "4571172                 chr1  ncbiRefSeq.2022-10-28         exon  156591776   \n",
      "4571173                 chr1  ncbiRefSeq.2022-10-28         5UTR  156591776   \n",
      "4571174                 chr1  ncbiRefSeq.2022-10-28          CDS  156591805   \n",
      "4571175                 chr1  ncbiRefSeq.2022-10-28         exon  156592101   \n",
      "4571176                 chr1  ncbiRefSeq.2022-10-28          CDS  156592101   \n",
      "4571177                 chr1  ncbiRefSeq.2022-10-28         exon  156592365   \n",
      "4571178                 chr1  ncbiRefSeq.2022-10-28          CDS  156592365   \n",
      "4571179                 chr1  ncbiRefSeq.2022-10-28         exon  156592557   \n",
      "4571180                 chr1  ncbiRefSeq.2022-10-28          CDS  156592557   \n",
      "4571181                 chr1  ncbiRefSeq.2022-10-28         exon  156593408   \n",
      "4571182                 chr1  ncbiRefSeq.2022-10-28          CDS  156593408   \n",
      "4571183                 chr1  ncbiRefSeq.2022-10-28         exon  156593882   \n",
      "4571184                 chr1  ncbiRefSeq.2022-10-28          CDS  156593882   \n",
      "4571185                 chr1  ncbiRefSeq.2022-10-28         3UTR  156594085   \n",
      "4571186                 chr1  ncbiRefSeq.2022-10-28  start_codon  156591805   \n",
      "4571187                 chr1  ncbiRefSeq.2022-10-28   stop_codon  156594082   \n",
      "\n",
      "               end score strand frame gene_id  transcript_id gene_name  \\\n",
      "241841      152856     .      +     .    NAXE    NM_144772.3      NAXE   \n",
      "241842      150543     .      +     .    NAXE    NM_144772.3      NAXE   \n",
      "241843      150361     .      +     .    NAXE    NM_144772.3      NAXE   \n",
      "241844      150543     .      +     0    NAXE    NM_144772.3      NAXE   \n",
      "241845      150766     .      +     .    NAXE    NM_144772.3      NAXE   \n",
      "241846      150766     .      +     1    NAXE    NM_144772.3      NAXE   \n",
      "241847      151032     .      +     .    NAXE    NM_144772.3      NAXE   \n",
      "241848      151032     .      +     0    NAXE    NM_144772.3      NAXE   \n",
      "241849      151227     .      +     .    NAXE    NM_144772.3      NAXE   \n",
      "241850      151227     .      +     0    NAXE    NM_144772.3      NAXE   \n",
      "241851      152112     .      +     .    NAXE    NM_144772.3      NAXE   \n",
      "241852      152112     .      +     0    NAXE    NM_144772.3      NAXE   \n",
      "241853      152856     .      +     .    NAXE    NM_144772.3      NAXE   \n",
      "241854      152638     .      +     2    NAXE    NM_144772.3      NAXE   \n",
      "241855      152856     .      +     .    NAXE    NM_144772.3      NAXE   \n",
      "241856      150364     .      +     0    NAXE    NM_144772.3      NAXE   \n",
      "241857      152641     .      +     0    NAXE    NM_144772.3      NAXE   \n",
      "4571171  156594299     .      +     .    NAXE  NM_144772.3_2      NAXE   \n",
      "4571172  156591986     .      +     .    NAXE  NM_144772.3_2      NAXE   \n",
      "4571173  156591804     .      +     .    NAXE  NM_144772.3_2      NAXE   \n",
      "4571174  156591986     .      +     0    NAXE  NM_144772.3_2      NAXE   \n",
      "4571175  156592209     .      +     .    NAXE  NM_144772.3_2      NAXE   \n",
      "4571176  156592209     .      +     1    NAXE  NM_144772.3_2      NAXE   \n",
      "4571177  156592475     .      +     .    NAXE  NM_144772.3_2      NAXE   \n",
      "4571178  156592475     .      +     0    NAXE  NM_144772.3_2      NAXE   \n",
      "4571179  156592670     .      +     .    NAXE  NM_144772.3_2      NAXE   \n",
      "4571180  156592670     .      +     0    NAXE  NM_144772.3_2      NAXE   \n",
      "4571181  156593555     .      +     .    NAXE  NM_144772.3_2      NAXE   \n",
      "4571182  156593555     .      +     0    NAXE  NM_144772.3_2      NAXE   \n",
      "4571183  156594299     .      +     .    NAXE  NM_144772.3_2      NAXE   \n",
      "4571184  156594081     .      +     2    NAXE  NM_144772.3_2      NAXE   \n",
      "4571185  156594299     .      +     .    NAXE  NM_144772.3_2      NAXE   \n",
      "4571186  156591807     .      +     0    NAXE  NM_144772.3_2      NAXE   \n",
      "4571187  156594084     .      +     0    NAXE  NM_144772.3_2      NAXE   \n",
      "\n",
      "        exon_number          exon_id  \n",
      "241841          NaN              NaN  \n",
      "241842            1    NM_144772.3.1  \n",
      "241843            1    NM_144772.3.1  \n",
      "241844            1    NM_144772.3.1  \n",
      "241845            2    NM_144772.3.2  \n",
      "241846            2    NM_144772.3.2  \n",
      "241847            3    NM_144772.3.3  \n",
      "241848            3    NM_144772.3.3  \n",
      "241849            4    NM_144772.3.4  \n",
      "241850            4    NM_144772.3.4  \n",
      "241851            5    NM_144772.3.5  \n",
      "241852            5    NM_144772.3.5  \n",
      "241853            6    NM_144772.3.6  \n",
      "241854            6    NM_144772.3.6  \n",
      "241855            6    NM_144772.3.6  \n",
      "241856            1    NM_144772.3.1  \n",
      "241857            6    NM_144772.3.6  \n",
      "4571171         NaN              NaN  \n",
      "4571172           1  NM_144772.3_2.1  \n",
      "4571173           1  NM_144772.3_2.1  \n",
      "4571174           1  NM_144772.3_2.1  \n",
      "4571175           2  NM_144772.3_2.2  \n",
      "4571176           2  NM_144772.3_2.2  \n",
      "4571177           3  NM_144772.3_2.3  \n",
      "4571178           3  NM_144772.3_2.3  \n",
      "4571179           4  NM_144772.3_2.4  \n",
      "4571180           4  NM_144772.3_2.4  \n",
      "4571181           5  NM_144772.3_2.5  \n",
      "4571182           5  NM_144772.3_2.5  \n",
      "4571183           6  NM_144772.3_2.6  \n",
      "4571184           6  NM_144772.3_2.6  \n",
      "4571185           6  NM_144772.3_2.6  \n",
      "4571186           1  NM_144772.3_2.1  \n",
      "4571187           6  NM_144772.3_2.6  \n",
      "\n",
      "Gene statistics:\n",
      "{'transcripts': 2, 'exons': 12, 'cds': 12, 'utrs': 4, 'chromosomes': ['chr1_MU273335v1_fix', 'chr1'], 'total_span': np.int64(156443967)}\n",
      "Available transcripts:\n",
      "         transcript_id           chromosome      start        end strand\n",
      "241841     NM_144772.3  chr1_MU273335v1_fix     150333     152856      +\n",
      "4571171  NM_144772.3_2                 chr1  156591776  156594299      +\n"
     ]
    }
   ],
   "source": [
    "from swissisoform.genome import GenomeHandler\n",
    "from swissisoform.visualize import GenomeVisualizer\n",
    "from swissisoform.isoform import AlternativeIsoform\n",
    "from swissisoform.mutations import MutationHandler\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "# Initialize the genome handler\n",
    "genome = GenomeHandler(\n",
    "   '../data/genome_data/hg38.fa',\n",
    "   '../data/genome_data/hg38.ncbiRefSeq.gtf'\n",
    ")\n",
    "\n",
    "# Get NAXE features\n",
    "naxe_features = genome.find_gene_features('NAXE')\n",
    "print(\"NAXE features:\")\n",
    "print(naxe_features)\n",
    "\n",
    "# Get gene statistics\n",
    "stats = genome.get_gene_stats('NAXE')\n",
    "print(\"\\nGene statistics:\")\n",
    "print(stats)\n",
    "\n",
    "# Get available transcript IDs for NAXE\n",
    "transcript_info = genome.get_transcript_ids('NAXE')\n",
    "print(\"Available transcripts:\")\n",
    "print(transcript_info)\n",
    "\n",
    "# Select a transcript ID (for example, the last one)\n",
    "transcript_id = transcript_info.iloc[-1]['transcript_id']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Initialize handlers\n",
    "alt_isoforms = AlternativeIsoform()\n",
    "\n",
    "# Load BED file\n",
    "alt_isoforms.load_bed('../data/ribosome_profiling/RiboTISHV6_Ly2024_AnnoToTruncation_exonintersect.bed')\n",
    "\n",
    "# Get the alternative isoform features\n",
    "alt_features = alt_isoforms.get_visualization_features('NAXE')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Initialize the visualizer\n",
    "visualizer = GenomeVisualizer(genome)\n",
    "\n",
    "# Visualize the gene without the alternative isoform\n",
    "visualizer.visualize_transcript('NAXE', transcript_id, alt_features=None, output_file=f'naxe_transcript_{transcript_id}.png')\n",
    "\n",
    "# Visualize the gene with the alternative isoform\n",
    "visualizer.visualize_transcript('NAXE', transcript_id, alt_features=alt_features, output_file=f'naxe_transcript_alt_{transcript_id}.png')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total gnomad variants: 2202\n",
      "gnomAD summary:\n",
      "{'total_variants': 2202, 'consequence_types': {'missense_variant': 646, 'intron_variant': 569, '5_prime_UTR_variant': 285, '3_prime_UTR_variant': 272, 'synonymous_variant': 187, 'frameshift_variant': 65, 'splice_region_variant': 57, 'stop_gained': 44, 'non_coding_transcript_exon_variant': 21, 'inframe_deletion': 13, 'splice_acceptor_variant': 12, 'splice_donor_variant': 11, 'stop_lost': 10, 'start_lost': 5, 'inframe_insertion': 3, 'stop_retained_variant': 2}, 'mean_allele_frequency': np.float64(0.004228829187649131), 'variants_by_impact': {'with_hgvsc': np.int64(2196), 'with_hgvsp': np.int64(984)}}\n"
     ]
    }
   ],
   "source": [
    "# Initialize the handler\n",
    "mutation_handler = MutationHandler()\n",
    "\n",
    "# Define the gene you want to analyze\n",
    "gene_name = \"NAXE\"\n",
    "\n",
    "# Get the mutation data\n",
    "gnomad_data = await mutation_handler.fetch_gnomad_data(gene_name)\n",
    "\n",
    "# Get all variants (no filter)\n",
    "gnomad_variants = mutation_handler.process_gnomad_variants(gnomad_data)\n",
    "print(\"Total gnomad variants:\", len(gnomad_variants))\n",
    "\n",
    "# Get gnomAD summary\n",
    "gnomad_summary = await mutation_handler.get_gnomad_summary(gene_name)\n",
    "print(\"gnomAD summary:\")\n",
    "print(gnomad_summary)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>position</th>\n",
       "      <th>variant_id</th>\n",
       "      <th>reference</th>\n",
       "      <th>alternate</th>\n",
       "      <th>consequence</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>transcript_version</th>\n",
       "      <th>hgvs</th>\n",
       "      <th>hgvsc</th>\n",
       "      <th>hgvsp</th>\n",
       "      <th>...</th>\n",
       "      <th>eas_ac_hom</th>\n",
       "      <th>eas_ac_hemi</th>\n",
       "      <th>afr_ac</th>\n",
       "      <th>afr_an</th>\n",
       "      <th>afr_ac_hom</th>\n",
       "      <th>afr_ac_hemi</th>\n",
       "      <th>ami_ac</th>\n",
       "      <th>ami_an</th>\n",
       "      <th>ami_ac_hom</th>\n",
       "      <th>ami_ac_hemi</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>156591753</td>\n",
       "      <td>1-156591753-CGCACATGCGCCG-C</td>\n",
       "      <td>CGCACATGCGCCG</td>\n",
       "      <td>C</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>ENST00000368234</td>\n",
       "      <td>7</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>16462</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>156591754</td>\n",
       "      <td>1-156591754-GCACATGC-G</td>\n",
       "      <td>GCACATGC</td>\n",
       "      <td>G</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>ENST00000679913</td>\n",
       "      <td>1</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>24502</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>538.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>156591756</td>\n",
       "      <td>1-156591756-A-C</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>ENST00000679913</td>\n",
       "      <td>1</td>\n",
       "      <td>n.1A&gt;C</td>\n",
       "      <td>n.1A&gt;C</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2374</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>156591756</td>\n",
       "      <td>1-156591756-A-G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>ENST00000679913</td>\n",
       "      <td>1</td>\n",
       "      <td>n.1A&gt;G</td>\n",
       "      <td>n.1A&gt;G</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2374</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>156591757</td>\n",
       "      <td>1-156591757-C-G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>ENST00000679913</td>\n",
       "      <td>1</td>\n",
       "      <td>n.2C&gt;G</td>\n",
       "      <td>n.2C&gt;G</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2028</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2197</th>\n",
       "      <td>156601761</td>\n",
       "      <td>1-156601761-A-G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>ENST00000680269</td>\n",
       "      <td>1</td>\n",
       "      <td>c.*269A&gt;G</td>\n",
       "      <td>c.*269A&gt;G</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>41444</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>912.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2198</th>\n",
       "      <td>156601761</td>\n",
       "      <td>1-156601761-A-C</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>ENST00000680269</td>\n",
       "      <td>1</td>\n",
       "      <td>c.*269A&gt;C</td>\n",
       "      <td>c.*269A&gt;C</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>41444</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>912.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2199</th>\n",
       "      <td>156601779</td>\n",
       "      <td>1-156601779-G-A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>ENST00000680269</td>\n",
       "      <td>1</td>\n",
       "      <td>c.*287G&gt;A</td>\n",
       "      <td>c.*287G&gt;A</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>114</td>\n",
       "      <td>41538</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>912.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2200</th>\n",
       "      <td>156601797</td>\n",
       "      <td>1-156601797-T-A</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>ENST00000680269</td>\n",
       "      <td>1</td>\n",
       "      <td>c.*305T&gt;A</td>\n",
       "      <td>c.*305T&gt;A</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>41146</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>912.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2201</th>\n",
       "      <td>156601801</td>\n",
       "      <td>1-156601801-A-G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>ENST00000680269</td>\n",
       "      <td>1</td>\n",
       "      <td>c.*309A&gt;G</td>\n",
       "      <td>c.*309A&gt;G</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>40890</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>912.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2202 rows × 58 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       position                   variant_id      reference alternate  \\\n",
       "0     156591753  1-156591753-CGCACATGCGCCG-C  CGCACATGCGCCG         C   \n",
       "1     156591754       1-156591754-GCACATGC-G       GCACATGC         G   \n",
       "2     156591756              1-156591756-A-C              A         C   \n",
       "3     156591756              1-156591756-A-G              A         G   \n",
       "4     156591757              1-156591757-C-G              C         G   \n",
       "...         ...                          ...            ...       ...   \n",
       "2197  156601761              1-156601761-A-G              A         G   \n",
       "2198  156601761              1-156601761-A-C              A         C   \n",
       "2199  156601779              1-156601779-G-A              G         A   \n",
       "2200  156601797              1-156601797-T-A              T         A   \n",
       "2201  156601801              1-156601801-A-G              A         G   \n",
       "\n",
       "                             consequence    transcript_id transcript_version  \\\n",
       "0                    5_prime_UTR_variant  ENST00000368234                  7   \n",
       "1     non_coding_transcript_exon_variant  ENST00000679913                  1   \n",
       "2     non_coding_transcript_exon_variant  ENST00000679913                  1   \n",
       "3     non_coding_transcript_exon_variant  ENST00000679913                  1   \n",
       "4     non_coding_transcript_exon_variant  ENST00000679913                  1   \n",
       "...                                  ...              ...                ...   \n",
       "2197                 3_prime_UTR_variant  ENST00000680269                  1   \n",
       "2198                 3_prime_UTR_variant  ENST00000680269                  1   \n",
       "2199                 3_prime_UTR_variant  ENST00000680269                  1   \n",
       "2200                 3_prime_UTR_variant  ENST00000680269                  1   \n",
       "2201                 3_prime_UTR_variant  ENST00000680269                  1   \n",
       "\n",
       "           hgvs      hgvsc hgvsp  ... eas_ac_hom  eas_ac_hemi  afr_ac  afr_an  \\\n",
       "0          None       None  None  ...          0            0       0   16462   \n",
       "1          None       None  None  ...          0            0       0   24502   \n",
       "2        n.1A>C     n.1A>C  None  ...          0            0       0    2374   \n",
       "3        n.1A>G     n.1A>G  None  ...          0            0       0    2374   \n",
       "4        n.2C>G     n.2C>G  None  ...          0            0       0    2028   \n",
       "...         ...        ...   ...  ...        ...          ...     ...     ...   \n",
       "2197  c.*269A>G  c.*269A>G  None  ...          0            0       0   41444   \n",
       "2198  c.*269A>C  c.*269A>C  None  ...          0            0       1   41444   \n",
       "2199  c.*287G>A  c.*287G>A  None  ...          0            0     114   41538   \n",
       "2200  c.*305T>A  c.*305T>A  None  ...          0            0       1   41146   \n",
       "2201  c.*309A>G  c.*309A>G  None  ...          0            0       0   40890   \n",
       "\n",
       "      afr_ac_hom  afr_ac_hemi ami_ac ami_an  ami_ac_hom  ami_ac_hemi  \n",
       "0              0            0    NaN    NaN         NaN          NaN  \n",
       "1              0            0    0.0  538.0         0.0          0.0  \n",
       "2              0            0    NaN    NaN         NaN          NaN  \n",
       "3              0            0    NaN    NaN         NaN          NaN  \n",
       "4              0            0    NaN    NaN         NaN          NaN  \n",
       "...          ...          ...    ...    ...         ...          ...  \n",
       "2197           0            0    0.0  912.0         0.0          0.0  \n",
       "2198           0            0    0.0  912.0         0.0          0.0  \n",
       "2199           0            0    0.0  912.0         0.0          0.0  \n",
       "2200           0            0    0.0  912.0         0.0          0.0  \n",
       "2201           0            0    0.0  912.0         0.0          0.0  \n",
       "\n",
       "[2202 rows x 58 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gnomad_variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/lab/barcheese01/mdiberna/swissisoform/src/swissisoform/mutations.py:390: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df[col] = pd.to_numeric(df[col], errors='ignore')\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total ClinVar variants: 142\n",
      "\n",
      "ClinVar summary:\n",
      "{'total_variants': 142, 'clinical_significance': {'Uncertain significance': 61, 'Likely benign': 35, 'Benign': 25, 'Pathogenic': 15, 'Likely pathogenic': 3, 'Conflicting classifications of pathogenicity': 2, 'Benign/Likely benign': 1}, 'variant_types': {'single nucleotide variant': 126, 'Deletion': 5, 'Indel': 4, 'Microsatellite': 3, 'Duplication': 2, 'Inversion': 1, 'Insertion': 1}, 'molecular_consequences': {'missense variant': 60, 'synonymous variant': 32, 'intron variant': 25, 'nonsense': 8, 'frameshift variant': 7, '': 5, 'inframe_deletion': 2, '3 prime UTR variant': 2, 'initiator_codon_variant': 1, 'splice donor variant': 1}, 'chromosome_distribution': {'1': 141, '': 1}, 'review_status': {'criteria provided, single submitter': 105, 'criteria provided, multiple submitters, no conflicts': 26, 'no assertion criteria provided': 9, 'criteria provided, conflicting classifications': 2}, 'metadata': {'submission_count': {'total': np.int64(194), 'mean': np.float64(1.37), 'max': np.int64(5)}, 'unique_accessions': 142}, 'pathogenicity_categories': {'likely_pathogenic_or_pathogenic': 18, 'benign_or_likely_benign': 61, 'uncertain_significance': 63, 'not_provided': 0}, 'location_stats': {'min_position': 156591622, 'max_position': 156594200, 'span': 2578}}\n"
     ]
    }
   ],
   "source": [
    "# Initialize the handler\n",
    "mutation_handler = MutationHandler()\n",
    "\n",
    "# Define the gene you want to analyze\n",
    "gene_name = \"NAXE\"\n",
    "\n",
    "# Get ClinVar data\n",
    "clinvar_variants = await mutation_handler.get_clinvar_variants(gene_name)\n",
    "print(\"Total ClinVar variants:\", len(clinvar_variants))\n",
    "\n",
    "# Get ClinVar summary\n",
    "clinvar_summary = await mutation_handler.get_clinvar_summary(gene_name)\n",
    "print(\"\\nClinVar summary:\")\n",
    "print(clinvar_summary)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>variant_id</th>\n",
       "      <th>obj_type</th>\n",
       "      <th>accession</th>\n",
       "      <th>title</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>review_status</th>\n",
       "      <th>clinical_significance</th>\n",
       "      <th>last_evaluated</th>\n",
       "      <th>molecular_consequences</th>\n",
       "      <th>protein_change</th>\n",
       "      <th>...</th>\n",
       "      <th>stop</th>\n",
       "      <th>ref_allele</th>\n",
       "      <th>alt_allele</th>\n",
       "      <th>submission_count</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_strand</th>\n",
       "      <th>variant_name</th>\n",
       "      <th>cdna_change</th>\n",
       "      <th>canonical_spdi</th>\n",
       "      <th>cytogenic_location</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3381841</td>\n",
       "      <td>Microsatellite</td>\n",
       "      <td>VCV003381841</td>\n",
       "      <td>NAXE, (GGGCC)n REPEAT EXPANSION, PROMOTER</td>\n",
       "      <td>NAXE</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>2024/11/26 00:00</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>128240</td>\n",
       "      <td>+</td>\n",
       "      <td>NAXE, (GGGCC)n REPEAT EXPANSION, PROMOTER</td>\n",
       "      <td>NAXE, (GGGCC)n REPEAT EXPANSION, PROMOTER</td>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3177706</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>VCV003177706</td>\n",
       "      <td>NM_144772.3(NAXE):c.589C&gt;T (p.Arg197Trp)</td>\n",
       "      <td>NAXE</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>Uncertain significance</td>\n",
       "      <td>2021/01/13 00:00</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>R197W</td>\n",
       "      <td>...</td>\n",
       "      <td>156593480.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>128240</td>\n",
       "      <td>+</td>\n",
       "      <td>NM_144772.3(NAXE):c.589C&gt;T (p.Arg197Trp)</td>\n",
       "      <td>c.589C&gt;T</td>\n",
       "      <td>NC_000001.11:156593479:C:T</td>\n",
       "      <td>1q22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3030803</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>VCV003030803</td>\n",
       "      <td>NM_144772.3(NAXE):c.183-6C&gt;T</td>\n",
       "      <td>NAXE</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>Likely benign</td>\n",
       "      <td>2022/03/16 00:00</td>\n",
       "      <td>intron variant</td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td>156592095.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>128240</td>\n",
       "      <td>+</td>\n",
       "      <td>NM_144772.3(NAXE):c.183-6C&gt;T</td>\n",
       "      <td>NM_144772.3(NAXE):c.183-6C&gt;T</td>\n",
       "      <td>NC_000001.11:156592094:C:T</td>\n",
       "      <td>1q22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3003683</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>VCV003003683</td>\n",
       "      <td>NM_144772.3(NAXE):c.862C&gt;T (p.Gln288Ter)</td>\n",
       "      <td>NAXE</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>Uncertain significance</td>\n",
       "      <td>2023/09/05 00:00</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>Q288*</td>\n",
       "      <td>...</td>\n",
       "      <td>156594079.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>128240</td>\n",
       "      <td>+</td>\n",
       "      <td>NM_144772.3(NAXE):c.862C&gt;T (p.Gln288Ter)</td>\n",
       "      <td>c.862C&gt;T</td>\n",
       "      <td>NC_000001.11:156594078:C:T</td>\n",
       "      <td>1q22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3001761</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>VCV003001761</td>\n",
       "      <td>NM_144772.3(NAXE):c.807G&gt;A (p.Lys269=)</td>\n",
       "      <td>NAXE</td>\n",
       "      <td>criteria provided, single submitter</td>\n",
       "      <td>Likely benign</td>\n",
       "      <td>2023/08/17 00:00</td>\n",
       "      <td>synonymous variant</td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td>156594024.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>128240</td>\n",
       "      <td>+</td>\n",
       "      <td>NM_144772.3(NAXE):c.807G&gt;A (p.Lys269=)</td>\n",
       "      <td>c.807G&gt;A</td>\n",
       "      <td>NC_000001.11:156594023:G:A</td>\n",
       "      <td>1q22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>137</th>\n",
       "      <td>268119</td>\n",
       "      <td>Indel</td>\n",
       "      <td>VCV000268119</td>\n",
       "      <td>NM_144772.3(NAXE):c.804_807delinsA (p.Lys270del)</td>\n",
       "      <td>NAXE</td>\n",
       "      <td>criteria provided, conflicting classifications</td>\n",
       "      <td>Conflicting classifications of pathogenicity</td>\n",
       "      <td>2024/05/16 00:00</td>\n",
       "      <td>inframe_deletion</td>\n",
       "      <td>K270del</td>\n",
       "      <td>...</td>\n",
       "      <td>156594024.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>4</td>\n",
       "      <td>128240</td>\n",
       "      <td>+</td>\n",
       "      <td>NM_144772.3(NAXE):c.804_807delinsA (p.Lys270del)</td>\n",
       "      <td>c.804_807delinsA</td>\n",
       "      <td>NC_000001.11:156594020:GAAG:A</td>\n",
       "      <td>1q22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>138</th>\n",
       "      <td>268118</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>VCV000268118</td>\n",
       "      <td>NM_144772.3(NAXE):c.516+1G&gt;A</td>\n",
       "      <td>NAXE</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>2016/11/14 00:00</td>\n",
       "      <td>splice donor variant</td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td>156592671.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>128240</td>\n",
       "      <td>+</td>\n",
       "      <td>NM_144772.3(NAXE):c.516+1G&gt;A</td>\n",
       "      <td>NM_144772.3(NAXE):c.516+1G&gt;A</td>\n",
       "      <td>NC_000001.11:156592670:G:A</td>\n",
       "      <td>1q22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139</th>\n",
       "      <td>268117</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>VCV000268117</td>\n",
       "      <td>NM_144772.3(NAXE):c.196C&gt;T (p.Gln66Ter)</td>\n",
       "      <td>NAXE</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>2016/11/14 00:00</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>Q66*</td>\n",
       "      <td>...</td>\n",
       "      <td>156592114.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>128240</td>\n",
       "      <td>+</td>\n",
       "      <td>NM_144772.3(NAXE):c.196C&gt;T (p.Gln66Ter)</td>\n",
       "      <td>c.196C&gt;T</td>\n",
       "      <td>NC_000001.11:156592113:C:T</td>\n",
       "      <td>1q22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140</th>\n",
       "      <td>268116</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>VCV000268116</td>\n",
       "      <td>NM_144772.3(NAXE):c.177C&gt;A (p.Tyr59Ter)</td>\n",
       "      <td>NAXE</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>2016/11/14 00:00</td>\n",
       "      <td>nonsense</td>\n",
       "      <td>Y59*</td>\n",
       "      <td>...</td>\n",
       "      <td>156591981.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>128240</td>\n",
       "      <td>+</td>\n",
       "      <td>NM_144772.3(NAXE):c.177C&gt;A (p.Tyr59Ter)</td>\n",
       "      <td>c.177C&gt;A</td>\n",
       "      <td>NC_000001.11:156591980:C:A</td>\n",
       "      <td>1q22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>141</th>\n",
       "      <td>253145</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>VCV000253145</td>\n",
       "      <td>NM_144772.3(NAXE):c.281C&gt;A (p.Ala94Asp)</td>\n",
       "      <td>NAXE</td>\n",
       "      <td>no assertion criteria provided</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>2016/11/14 00:00</td>\n",
       "      <td>missense variant</td>\n",
       "      <td>A94D</td>\n",
       "      <td>...</td>\n",
       "      <td>156592199.0</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>1</td>\n",
       "      <td>128240</td>\n",
       "      <td>+</td>\n",
       "      <td>NM_144772.3(NAXE):c.281C&gt;A (p.Ala94Asp)</td>\n",
       "      <td>c.281C&gt;A</td>\n",
       "      <td>NC_000001.11:156592198:C:A</td>\n",
       "      <td>1q22</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>142 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    variant_id                   obj_type     accession  \\\n",
       "0      3381841             Microsatellite  VCV003381841   \n",
       "1      3177706  single nucleotide variant  VCV003177706   \n",
       "2      3030803  single nucleotide variant  VCV003030803   \n",
       "3      3003683  single nucleotide variant  VCV003003683   \n",
       "4      3001761  single nucleotide variant  VCV003001761   \n",
       "..         ...                        ...           ...   \n",
       "137     268119                      Indel  VCV000268119   \n",
       "138     268118  single nucleotide variant  VCV000268118   \n",
       "139     268117  single nucleotide variant  VCV000268117   \n",
       "140     268116  single nucleotide variant  VCV000268116   \n",
       "141     253145  single nucleotide variant  VCV000253145   \n",
       "\n",
       "                                                title gene_name  \\\n",
       "0           NAXE, (GGGCC)n REPEAT EXPANSION, PROMOTER      NAXE   \n",
       "1            NM_144772.3(NAXE):c.589C>T (p.Arg197Trp)      NAXE   \n",
       "2                        NM_144772.3(NAXE):c.183-6C>T      NAXE   \n",
       "3            NM_144772.3(NAXE):c.862C>T (p.Gln288Ter)      NAXE   \n",
       "4              NM_144772.3(NAXE):c.807G>A (p.Lys269=)      NAXE   \n",
       "..                                                ...       ...   \n",
       "137  NM_144772.3(NAXE):c.804_807delinsA (p.Lys270del)      NAXE   \n",
       "138                      NM_144772.3(NAXE):c.516+1G>A      NAXE   \n",
       "139           NM_144772.3(NAXE):c.196C>T (p.Gln66Ter)      NAXE   \n",
       "140           NM_144772.3(NAXE):c.177C>A (p.Tyr59Ter)      NAXE   \n",
       "141           NM_144772.3(NAXE):c.281C>A (p.Ala94Asp)      NAXE   \n",
       "\n",
       "                                      review_status  \\\n",
       "0                    no assertion criteria provided   \n",
       "1               criteria provided, single submitter   \n",
       "2                    no assertion criteria provided   \n",
       "3               criteria provided, single submitter   \n",
       "4               criteria provided, single submitter   \n",
       "..                                              ...   \n",
       "137  criteria provided, conflicting classifications   \n",
       "138                  no assertion criteria provided   \n",
       "139                  no assertion criteria provided   \n",
       "140                  no assertion criteria provided   \n",
       "141                  no assertion criteria provided   \n",
       "\n",
       "                            clinical_significance    last_evaluated  \\\n",
       "0                                      Pathogenic  2024/11/26 00:00   \n",
       "1                          Uncertain significance  2021/01/13 00:00   \n",
       "2                                   Likely benign  2022/03/16 00:00   \n",
       "3                          Uncertain significance  2023/09/05 00:00   \n",
       "4                                   Likely benign  2023/08/17 00:00   \n",
       "..                                            ...               ...   \n",
       "137  Conflicting classifications of pathogenicity  2024/05/16 00:00   \n",
       "138                                    Pathogenic  2016/11/14 00:00   \n",
       "139                                    Pathogenic  2016/11/14 00:00   \n",
       "140                                    Pathogenic  2016/11/14 00:00   \n",
       "141                                    Pathogenic  2016/11/14 00:00   \n",
       "\n",
       "    molecular_consequences protein_change  ...         stop  ref_allele  \\\n",
       "0                                          ...          NaN               \n",
       "1         missense variant          R197W  ...  156593480.0               \n",
       "2           intron variant                 ...  156592095.0               \n",
       "3                 nonsense          Q288*  ...  156594079.0               \n",
       "4       synonymous variant                 ...  156594024.0               \n",
       "..                     ...            ...  ...          ...         ...   \n",
       "137       inframe_deletion        K270del  ...  156594024.0               \n",
       "138   splice donor variant                 ...  156592671.0               \n",
       "139               nonsense           Q66*  ...  156592114.0               \n",
       "140               nonsense           Y59*  ...  156591981.0               \n",
       "141       missense variant           A94D  ...  156592199.0               \n",
       "\n",
       "     alt_allele submission_count gene_id  gene_strand  \\\n",
       "0                              1  128240            +   \n",
       "1                              1  128240            +   \n",
       "2                              1  128240            +   \n",
       "3                              1  128240            +   \n",
       "4                              1  128240            +   \n",
       "..          ...              ...     ...          ...   \n",
       "137                            4  128240            +   \n",
       "138                            1  128240            +   \n",
       "139                            1  128240            +   \n",
       "140                            1  128240            +   \n",
       "141                            1  128240            +   \n",
       "\n",
       "                                         variant_name  \\\n",
       "0           NAXE, (GGGCC)n REPEAT EXPANSION, PROMOTER   \n",
       "1            NM_144772.3(NAXE):c.589C>T (p.Arg197Trp)   \n",
       "2                        NM_144772.3(NAXE):c.183-6C>T   \n",
       "3            NM_144772.3(NAXE):c.862C>T (p.Gln288Ter)   \n",
       "4              NM_144772.3(NAXE):c.807G>A (p.Lys269=)   \n",
       "..                                                ...   \n",
       "137  NM_144772.3(NAXE):c.804_807delinsA (p.Lys270del)   \n",
       "138                      NM_144772.3(NAXE):c.516+1G>A   \n",
       "139           NM_144772.3(NAXE):c.196C>T (p.Gln66Ter)   \n",
       "140           NM_144772.3(NAXE):c.177C>A (p.Tyr59Ter)   \n",
       "141           NM_144772.3(NAXE):c.281C>A (p.Ala94Asp)   \n",
       "\n",
       "                                   cdna_change                 canonical_spdi  \\\n",
       "0    NAXE, (GGGCC)n REPEAT EXPANSION, PROMOTER                                  \n",
       "1                                     c.589C>T     NC_000001.11:156593479:C:T   \n",
       "2                 NM_144772.3(NAXE):c.183-6C>T     NC_000001.11:156592094:C:T   \n",
       "3                                     c.862C>T     NC_000001.11:156594078:C:T   \n",
       "4                                     c.807G>A     NC_000001.11:156594023:G:A   \n",
       "..                                         ...                            ...   \n",
       "137                           c.804_807delinsA  NC_000001.11:156594020:GAAG:A   \n",
       "138               NM_144772.3(NAXE):c.516+1G>A     NC_000001.11:156592670:G:A   \n",
       "139                                   c.196C>T     NC_000001.11:156592113:C:T   \n",
       "140                                   c.177C>A     NC_000001.11:156591980:C:A   \n",
       "141                                   c.281C>A     NC_000001.11:156592198:C:A   \n",
       "\n",
       "    cytogenic_location  \n",
       "0                  NaN  \n",
       "1                 1q22  \n",
       "2                 1q22  \n",
       "3                 1q22  \n",
       "4                 1q22  \n",
       "..                 ...  \n",
       "137               1q22  \n",
       "138               1q22  \n",
       "139               1q22  \n",
       "140               1q22  \n",
       "141               1q22  \n",
       "\n",
       "[142 rows x 22 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total aggregator variants: 383\n",
      "Aggregator summary: {'csv_path': '../data/mutation_data/NAXE.csv', 'total_variants': 383, 'impact_counts': {'MODIFIER': 321, 'MODERATE': 36, 'LOW': 24, 'HIGH': 2}, 'consequence_counts': {'downstream_gene_variant': 224, 'intron_variant': 56, 'missense_variant': 35, 'upstream_gene_variant': 26, 'synonymous_variant': 14, '3_prime_UTR_variant': 14, 'splice_polypyrimidine_tract_variant,intron_variant': 5, 'splice_region_variant,intron_variant': 2, '5_prime_UTR_variant': 1, 'missense_variant,splice_region_variant': 1, 'splice_donor_region_variant,intron_variant': 1, 'stop_gained': 1, 'splice_region_variant,synonymous_variant': 1, 'splice_donor_variant': 1, 'splice_region_variant,splice_polypyrimidine_tract_variant,intron_variant': 1}, 'position_stats': {'chromosomes': ['1'], 'min_position': 156591106, 'max_position': 156599294}, 'frequency_stats': {'mean_frequency': np.float64(0.028479332375979115), 'median_frequency': np.float64(9.28e-05), 'max_frequency': np.float64(0.992)}, 'notation_stats': {'with_hgvsc': np.int64(383), 'with_hgvsp': np.int64(383)}}\n"
     ]
    }
   ],
   "source": [
    "# Instantiate the handler\n",
    "mutation_handler = MutationHandler()\n",
    "\n",
    "# Specify the path to your aggregator CSV (e.g. NAXE.csv)\n",
    "aggregator_csv_path = \"../data/mutation_data/NAXE.csv\"\n",
    "\n",
    "# Get the aggregator variants DataFrame\n",
    "aggregator_variants = mutation_handler.get_aggregator_variants(aggregator_csv_path)\n",
    "print(\"Total aggregator variants:\", len(aggregator_variants))\n",
    "\n",
    "# Get a summary of the aggregator data\n",
    "aggregator_summary = mutation_handler.get_aggregator_summary(aggregator_csv_path)\n",
    "print(\"Aggregator summary:\", aggregator_summary)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>varId</th>\n",
       "      <th>impact</th>\n",
       "      <th>consequence</th>\n",
       "      <th>hgvsc</th>\n",
       "      <th>hgvsp</th>\n",
       "      <th>allelecount</th>\n",
       "      <th>allelnumber</th>\n",
       "      <th>allelefrequency</th>\n",
       "      <th>homozygouscount</th>\n",
       "      <th>gnomADg_AC</th>\n",
       "      <th>gnomADg_AN</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>reference</th>\n",
       "      <th>alternate</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1:156591106:G:A</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>5</td>\n",
       "      <td>1732</td>\n",
       "      <td>0.002890</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>156591106</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1:156591114:G:C</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>9</td>\n",
       "      <td>1748</td>\n",
       "      <td>0.005150</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>156591114</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1:156591131:T:TAG</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>2</td>\n",
       "      <td>1748</td>\n",
       "      <td>0.001140</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>156591131</td>\n",
       "      <td>T</td>\n",
       "      <td>TAG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1:156591244:G:C</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>2</td>\n",
       "      <td>1900</td>\n",
       "      <td>0.001050</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>156591244</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1:156591269:T:C</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>1848</td>\n",
       "      <td>0.000541</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>156591269</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>378</th>\n",
       "      <td>1:156599232:G:A</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>2</td>\n",
       "      <td>21038</td>\n",
       "      <td>0.000095</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>156599232</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>379</th>\n",
       "      <td>1:156599271:C:T</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>2</td>\n",
       "      <td>15526</td>\n",
       "      <td>0.000129</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>156599271</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>380</th>\n",
       "      <td>1:156599285:G:T</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>20</td>\n",
       "      <td>15072</td>\n",
       "      <td>0.001330</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>156599285</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>381</th>\n",
       "      <td>1:156599291:A:C</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>31</td>\n",
       "      <td>14890</td>\n",
       "      <td>0.002080</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>156599291</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>382</th>\n",
       "      <td>1:156599294:G:A</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>2</td>\n",
       "      <td>14860</td>\n",
       "      <td>0.000135</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>156599294</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>383 rows × 16 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 varId    impact              consequence hgvsc hgvsp  \\\n",
       "0      1:156591106:G:A  MODIFIER    upstream_gene_variant     -     -   \n",
       "1      1:156591114:G:C  MODIFIER    upstream_gene_variant     -     -   \n",
       "2    1:156591131:T:TAG  MODIFIER    upstream_gene_variant     -     -   \n",
       "3      1:156591244:G:C  MODIFIER    upstream_gene_variant     -     -   \n",
       "4      1:156591269:T:C  MODIFIER    upstream_gene_variant     -     -   \n",
       "..                 ...       ...                      ...   ...   ...   \n",
       "378    1:156599232:G:A  MODIFIER  downstream_gene_variant     -     -   \n",
       "379    1:156599271:C:T  MODIFIER  downstream_gene_variant     -     -   \n",
       "380    1:156599285:G:T  MODIFIER  downstream_gene_variant     -     -   \n",
       "381    1:156599291:A:C  MODIFIER  downstream_gene_variant     -     -   \n",
       "382    1:156599294:G:A  MODIFIER  downstream_gene_variant     -     -   \n",
       "\n",
       "     allelecount  allelnumber  allelefrequency  homozygouscount  gnomADg_AC  \\\n",
       "0              5         1732         0.002890                0         NaN   \n",
       "1              9         1748         0.005150                2         NaN   \n",
       "2              2         1748         0.001140                0         NaN   \n",
       "3              2         1900         0.001050                0         NaN   \n",
       "4              1         1848         0.000541                0         NaN   \n",
       "..           ...          ...              ...              ...         ...   \n",
       "378            2        21038         0.000095                0         NaN   \n",
       "379            2        15526         0.000129                0         NaN   \n",
       "380           20        15072         0.001330                2         NaN   \n",
       "381           31        14890         0.002080                2         NaN   \n",
       "382            2        14860         0.000135                1         NaN   \n",
       "\n",
       "     gnomADg_AN  gnomADg_AF chromosome   position reference alternate  \n",
       "0           NaN         NaN          1  156591106         G         A  \n",
       "1           NaN         NaN          1  156591114         G         C  \n",
       "2           NaN         NaN          1  156591131         T       TAG  \n",
       "3           NaN         NaN          1  156591244         G         C  \n",
       "4           NaN         NaN          1  156591269         T         C  \n",
       "..          ...         ...        ...        ...       ...       ...  \n",
       "378         NaN         NaN          1  156599232         G         A  \n",
       "379         NaN         NaN          1  156599271         C         T  \n",
       "380         NaN         NaN          1  156599285         G         T  \n",
       "381         NaN         NaN          1  156599291         A         C  \n",
       "382         NaN         NaN          1  156599294         G         A  \n",
       "\n",
       "[383 rows x 16 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "aggregator_variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/lab/barcheese01/mdiberna/swissisoform/src/swissisoform/mutations.py:390: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df[col] = pd.to_numeric(df[col], errors='ignore')\n"
     ]
    }
   ],
   "source": [
    "# Initialize handler\n",
    "mutation_handler = MutationHandler()\n",
    "\n",
    "# Get data from all sources\n",
    "standardized_mutations = await mutation_handler.get_visualization_ready_mutations(\n",
    "    gene_name='NAXE',\n",
    "    aggregator_csv_path=\"../data/mutation_data/NAXE.csv\"\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>position</th>\n",
       "      <th>variant_id</th>\n",
       "      <th>reference</th>\n",
       "      <th>alternate</th>\n",
       "      <th>source</th>\n",
       "      <th>impact</th>\n",
       "      <th>hgvsc</th>\n",
       "      <th>hgvsp</th>\n",
       "      <th>allele_frequency</th>\n",
       "      <th>clinical_significance</th>\n",
       "      <th>relative_position</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>156591753.0</td>\n",
       "      <td>1-156591753-CGCACATGCGCCG-C</td>\n",
       "      <td>CGCACATGCGCCG</td>\n",
       "      <td>C</td>\n",
       "      <td>gnomAD</td>\n",
       "      <td>5_prime_UTR_variant</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>647.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>156591754.0</td>\n",
       "      <td>1-156591754-GCACATGC-G</td>\n",
       "      <td>GCACATGC</td>\n",
       "      <td>G</td>\n",
       "      <td>gnomAD</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.000011</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>648.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>156591756.0</td>\n",
       "      <td>1-156591756-A-G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>gnomAD</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>n.1A&gt;G</td>\n",
       "      <td>.</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>650.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>156591756.0</td>\n",
       "      <td>1-156591756-A-C</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>gnomAD</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>n.1A&gt;C</td>\n",
       "      <td>.</td>\n",
       "      <td>0.000028</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>650.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>156591757.0</td>\n",
       "      <td>1-156591757-C-A</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>gnomAD</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>n.2C&gt;A</td>\n",
       "      <td>.</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>651.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2722</th>\n",
       "      <td>156599232.0</td>\n",
       "      <td>1:156599232:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Aggregator</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.000095</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>8126.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2723</th>\n",
       "      <td>156599271.0</td>\n",
       "      <td>1:156599271:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Aggregator</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.000129</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>8165.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2724</th>\n",
       "      <td>156599285.0</td>\n",
       "      <td>1:156599285:G:T</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>Aggregator</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.001330</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>8179.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2725</th>\n",
       "      <td>156599291.0</td>\n",
       "      <td>1:156599291:A:C</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>Aggregator</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.002080</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>8185.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2726</th>\n",
       "      <td>156599294.0</td>\n",
       "      <td>1:156599294:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Aggregator</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.000135</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>8188.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2626 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         position                   variant_id      reference alternate  \\\n",
       "0     156591753.0  1-156591753-CGCACATGCGCCG-C  CGCACATGCGCCG         C   \n",
       "1     156591754.0       1-156591754-GCACATGC-G       GCACATGC         G   \n",
       "2     156591756.0              1-156591756-A-G              A         G   \n",
       "3     156591756.0              1-156591756-A-C              A         C   \n",
       "4     156591757.0              1-156591757-C-A              C         A   \n",
       "...           ...                          ...            ...       ...   \n",
       "2722  156599232.0              1:156599232:G:A              G         A   \n",
       "2723  156599271.0              1:156599271:C:T              C         T   \n",
       "2724  156599285.0              1:156599285:G:T              G         T   \n",
       "2725  156599291.0              1:156599291:A:C              A         C   \n",
       "2726  156599294.0              1:156599294:G:A              G         A   \n",
       "\n",
       "          source                              impact   hgvsc hgvsp  \\\n",
       "0         gnomAD                 5_prime_UTR_variant       .     .   \n",
       "1         gnomAD  non_coding_transcript_exon_variant       .     .   \n",
       "2         gnomAD  non_coding_transcript_exon_variant  n.1A>G     .   \n",
       "3         gnomAD  non_coding_transcript_exon_variant  n.1A>C     .   \n",
       "4         gnomAD  non_coding_transcript_exon_variant  n.2C>A     .   \n",
       "...          ...                                 ...     ...   ...   \n",
       "2722  Aggregator             downstream_gene_variant       -     -   \n",
       "2723  Aggregator             downstream_gene_variant       -     -   \n",
       "2724  Aggregator             downstream_gene_variant       -     -   \n",
       "2725  Aggregator             downstream_gene_variant       -     -   \n",
       "2726  Aggregator             downstream_gene_variant       -     -   \n",
       "\n",
       "      allele_frequency clinical_significance  relative_position  \n",
       "0             0.000001               Unknown              647.0  \n",
       "1             0.000011               Unknown              648.0  \n",
       "2             0.000000               Unknown              650.0  \n",
       "3             0.000028               Unknown              650.0  \n",
       "4             0.000000               Unknown              651.0  \n",
       "...                ...                   ...                ...  \n",
       "2722          0.000095               Unknown             8126.0  \n",
       "2723          0.000129               Unknown             8165.0  \n",
       "2724          0.001330               Unknown             8179.0  \n",
       "2725          0.002080               Unknown             8185.0  \n",
       "2726          0.000135               Unknown             8188.0  \n",
       "\n",
       "[2626 rows x 11 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "standardized_mutations"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "swissisoform",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.16"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
